mlir/include/mlir/Dialect/Func/Utils/Utils.h

```diff
@@ -0,0 +1,42 @@
+//===- Utils.h - General Func transformation utilities ----*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This header file defines prototypes for various transformation utilities for
+// the Func dialect. These are not passes by themselves but are used
+// either by passes, optimization sequences, or in turn by other transformation
+// utilities.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_DIALECT_FUNC_UTILS_H
+#define MLIR_DIALECT_FUNC_UTILS_H
+
+#include "mlir/Dialect/Func/IR/FuncOps.h"
```

Comment:
Nit: we prefer to forward-declare classes in headers instead of including whenever possible (shortens compile/preprocessing time).